internal/controller/operator/operandrequest.go

```diff
@@ -37,23 +37,32 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-// addEmbeddedEDBIfNeeded appends the common-service-postgresql Operand to the list of Operands when the IM install is
+// addEmbeddedDBIfNeeded appends the common-service-postgresql Operand to the list of Operands when the IM install is
 // configured to use an embedded EDB. If there is an error while trying to obtain the relevant IM configuration for EDB,
 // it will skip adding the Operand and return the encountered error.
-func (r *AuthenticationReconciler) addEmbeddedEDBIfNeeded(ctx context.Context, authCR *operatorv1alpha1.Authentication,
+func (r *AuthenticationReconciler) addEmbeddedDBIfNeeded(ctx context.Context, authCR *operatorv1alpha1.Authentication,
 	operands *[]operatorv1alpha1.Operand) (err error) {
 	var usingExternal bool
-	if usingExternal, err = r.isConfiguredForExternalEDB(ctx, authCR); err == nil && !usingExternal {
-		*operands = append(*operands, operatorv1alpha1.Operand{
-			Name: "common-service-postgresql",
-			Bindings: map[string]operatorv1alpha1.Bindable{
-				"protected-im-db": {
-					Secret:    ctrlcommon.DatastoreEDBSecretName,
-					Configmap: ctrlcommon.DatastoreEDBCMName,
-				},
-			},
-		})
+	if usingExternal, err = r.isConfiguredForExternalEDB(ctx, authCR); err != nil || usingExternal {
+		return
+	}
+	name := "common-service-postgresql"
+	if authCR.Spec.Database != nil && authCR.Spec.Database.OperandRequest != nil && *authCR.Spec.Database.OperandRequest != "" {
+		if *authCR.Spec.Database.OperandRequest == "ibm-cnpg-postgres-operator" {
+			name = "common-service-cnpg"
+		} else {
+			name = *authCR.Spec.Database.OperandRequest
```

Comment:
Hi @rwhundley , Just for my understanding. apart from `common-service-postgresql` and `common-service-cnpg` is there anything else that gets assigned to name with `*authCR.Spec.Database.OperandRequest`